### DISCUSSION TITLE
Detect If Client Has Server Connection

### DISCUSSION THREAD

**Discussion by keenanjohnson** [zauberzeug/nicegui#3270] [Q&A] [answered]
### Question

Is there a straightforward way to display on the GUI if a client has an activate connection to the nicegui server?

I'm building a control app for a piece of hardware and it's somewhat plausible that a power loss, etc could cause the server to go down and I would like to make it clear in the client page if that's the case through a popup etc.

Thanks!

**Comment by falkoschindler** [comment, marked as answer]
Hi @keenanjohnson,

By default NiceGUI is showing a popup when the server connection got lost. Is this what you mean?

<img width="588" alt="Screenshot 2024-06-23 at 17 32 14" src="https://github.com/zauberzeug/nicegui/assets/5767091/4ffd01ba-be98-4e45-a8f3-24eb00679f0d">

**Reply by keenanjohnson** [reply]
Hello @falkoschindler ! Yes I didn't realize that would happen.

Is there any way to configure the client to poll the connection status and display a positive connection confirmation as well? Ideally I would bind the status to an indicator and have the GUI.

**Reply by falkoschindler** [reply]
You could check the socket connection regularly using some custom JavaScript:
```py
ui.add_head_html('''
    <script>
        setInterval(() => {
            if (!window.socket.connected) {
                 alert('Socket.io connection lost');
            }
        }, 500);
    </script>
''')
```

**Reply by keenanjohnson** [reply]
Thanks for the reply @falkoschindler ! I'll try to figure out how to bind that to a storage item that I can use to map to an indicator in the header display of my application.
